src/Collection/UniqueObjectCollection.php

```diff
@@ -30,6 +30,18 @@ public function __construct(array $data)
 	}
 
 
+	/**
+	 * @param IUniqueObjectCollection $collection
+	 * @return static
+	 */
+	public function mergeWith(IUniqueObjectCollection $collection)
+	{
+		return new static(
+			array_replace($this->data, $collection->data)
```

Comment:
Just an idea: should be there some check for data type?
```php
if ($collection->getItemType() !== $this->getItemType()) {...}
```

Reply:
good point